opensquirrel/squirrel_ir.py

```diff
@@ -200,8 +199,11 @@ def __init__(self, control_qubit: Qubit, target_gate: Gate, generator=None, argu
         self.target_gate = target_gate
 
     def __eq__(self, other):
-        if not isinstance(other, ControlledGate):
+       
+        if not isinstance(other, ControlledGate | MatrixGate):
```

Comment:
All these `ControlledGate | MatrixGate` checks... aren't a bit of "magic code"? Could we use an interface class of which these two would inherit, and that would express a given feature that both types of gate have? Again, if a `ControlledGate` were a `MatrixGate`, the check `ControlledGate |` wouldn't be needed.

Reply:
Yes, this sounds a lot better. It could also open the door to importing controlled gates from other frameworks.